pebble_tool/commands/base.py

```diff
@@ -70,37 +70,28 @@ def _set_debugging(self, level):
 
 
 class PebbleCommand(BaseCommand):
-    valid_connections = {'phone', 'qemu', 'cloudpebble', 'emulator', 'serial'}
+    connection_handlers = set()
+
+    @classmethod
+    def register_connection_handler(cls, impl):
+        cls.connection_handlers.add(impl)
 
     @classmethod
     def _shared_parser(cls):
         parser = argparse.ArgumentParser(add_help=False)
-        if len(cls.valid_connections) < 2 :
-            group = parser
-        else :
-            group = parser.add_mutually_exclusive_group()
-        if 'phone' in cls.valid_connections:
-            group.add_argument('--phone', metavar='phone_ip',
-                                help="When using the developer connection, your phone's IP or hostname. "
-                                     "Equivalent to PEBBLE_PHONE.")
-        if 'qemu' in cls.valid_connections:
-            group.add_argument('--qemu', nargs='?', const='localhost:12344', metavar='host',
-                                help="Use this option to connect directly to a QEMU instance. "
-                                     "Equivalent to PEBBLE_QEMU.")
-        if 'cloudpebble' in cls.valid_connections:
-            group.add_argument('--cloudpebble', action='store_true', help="Use this option to connect to your phone via"
-                                                                          " the CloudPebble connection. Equivalent to "
-                                                                          "PEBBLE_CLOUDPEBBLE.")
-        if 'emulator' in cls.valid_connections:
-            emu_group = group.add_argument_group()
-            emu_group.add_argument('--emulator', type=str, help="Launch an emulator. Equivalent to PEBBLE_EMULATOR.",
-                               choices=pebble_platforms)
-            emu_group.add_argument('--sdk', type=str, help="SDK version to launch. Defaults to the active SDK"
-                                                       " (currently {})".format(sdk_version()))
-        if 'serial' in cls.valid_connections:
-            group.add_argument('--serial', type=str, help="Connected directly, given a path to a serial device.")
+        group = parser.add_mutually_exclusive_group()
+        for handler_impl in cls.valid_connection_handlers():
+            handler_impl.add_argument_handler(group)
         return super(PebbleCommand, cls)._shared_parser() + [parser]
 
+    @classmethod
+    def valid_connection_handlers(cls):
+        valid_connections = getattr(cls, 'valid_connections', None)
+        if valid_connections:
```

Comment:
Welp, yes. Rejigged this slightly before uploading from using hasattr and screwed up. Fixed!